recipes/qt/5.x.x/conanfile.py

```diff
@@ -0,0 +1,710 @@
+                return "macx-ios-clang"
+
+        elif self.settings.os == "watchOS":
+            if self.settings.compiler == "apple-clang":
+                return "macx-watchos-clang"
+
+        elif self.settings.os == "tvOS":
+            if self.settings.compiler == "apple-clang":
+                return "macx-tvos-clang"
+
+        elif self.settings.os == "Android":
+            if self.settings.compiler == "clang":
+                return "android-clang"
+
+        elif self.settings.os == "Windows":
+            return {"Visual Studio": "win32-msvc",
+                    "gcc": "win32-g++",
+                    "clang": "win32-clang-g++"}.get(str(self.settings.compiler))
+
+        elif self.settings.os == "WindowsStore":
+            if self.settings.compiler == "Visual Studio":
+                return {"14": {"armv7": "winrt-arm-msvc2015",
+                               "x86": "winrt-x86-msvc2015",
+                               "x86_64": "winrt-x64-msvc2015"},
+                        "15": {"armv7": "winrt-arm-msvc2017",
+                               "x86": "winrt-x86-msvc2017",
+                               "x86_64": "winrt-x64-msvc2017"}
+                        }.get(str(self.settings.compiler.version)).get(str(self.settings.arch))
+
+        elif self.settings.os == "FreeBSD":
+            return {"clang": "freebsd-clang",
+                    "gcc": "freebsd-g++"}.get(str(self.settings.compiler))
+
+        elif self.settings.os == "SunOS":
+            if self.settings.compiler == "sun-cc":
+                if self.settings.arch == "sparc":
+                    return "solaris-cc-stlport" if self.settings.compiler.libcxx == "libstlport" else "solaris-cc"
+                elif self.settings.arch == "sparcv9":
+                    return "solaris-cc64-stlport" if self.settings.compiler.libcxx == "libstlport" else "solaris-cc64"
+            elif self.settings.compiler == "gcc":
+                return {"sparc": "solaris-g++",
+                        "sparcv9": "solaris-g++-64"}.get(str(self.settings.arch))
+        elif self.settings.os == "Neutrino" and self.settings.compiler == "qcc":
+            return {"armv8": "qnx-aarch64le-qcc",
+                    "armv8.3": "qnx-aarch64le-qcc",
+                    "armv7": "qnx-armle-v7-qcc",
+                    "armv7hf": "qnx-armle-v7-qcc",
+                    "armv7s": "qnx-armle-v7-qcc",
+                    "armv7k": "qnx-armle-v7-qcc",
+                    "x86": "qnx-x86-qcc",
+                    "x86_64": "qnx-x86-64-qcc"}.get(str(self.settings.arch))
+        elif self.settings.os == "Emscripten" and self.settings.arch == "wasm":
+            return "wasm-emscripten"
+
+        return None
+
+    def build(self):
+        if self.settings.compiler in ["gcc", "clang"]:
+            if tools.Version(self.settings.compiler.version) < "5.0":
+                raise ConanInvalidConfiguration("qt 5.15.0 is not support on GCC or clang before 5.0")
```

Comment:
This would be another nit-picking thing: changing `Qt 5.15.0` to something like just `Qt 5`. We don't even offer 5.15.0 here :)

Reply:
I would leave as eventually we may add more versions

Reply:
The `if` applies to all Qt 5 versions, so the comment is wrong

Reply:
Should it be in `validate()`?